### ISSUE TITLE
Image Preview shown displaced because of Adblock

### ISSUE THREAD

**Issue by sersanor** [polimediaupv/paella#125]
![captura de pantalla 2015-03-23 a las 9 05 27](https://cloud.githubusercontent.com/assets/6085639/6776209/36b5ed00-d13c-11e4-8a50-c4aed65a0fde.png)

This, happens using Safari 8.03 with Adblock extension.

**Comment by djcp** [comment]
Nice catch! We've been bitten by this too.

**Comment by jjeising** [comment]
The issue here is that in some browsers the image takes a bit longer to load or until it is available in the DOM. paella uses the high of the element [here](https://github.com/polimediaupv/paella/blob/master/src/06_ui_controls.js#L198):
```javascript
//TOP ADJUSTO TO IMAGE RES
p = $("#divTimeImageOverlay").height();
$("#divTimeImageOverlay").css("top",pos.top-p);
```
If the image isn't loaded/completely available at this time the wrong position is calculated from the much smaller height.

A quick and not very beautiful fix is to add these lines after line 199:
```javascript
$("#imgOverlay").one('load', function(event) {
	var p = $("#divTimeImageOverlay").height();
	$("#divTimeImageOverlay").css("top",pos.top-p);
});
```
This will update the position again if the load event was fired and fix it for Safari.

**Comment by ferserc1** [comment]
I think that this is not the problem. This code is triggered when the mouse moves over the timeline bar. If the image is not loaded, the container will not be shown, otherwise the image will be correctly placed.

In fact, if you try to adjust the image position inside the load event handler, this event will never be triggered if the image is already loaded.

It's curious that this bug happen if you Safari has the adblock extension enabled, even though adblock is disabled in the player's page, but not if you disable adblock from Safari preferences. It should be something related with the adblock.

![captura de pantalla 2015-09-08 a las 13 26 39](https://cloud.githubusercontent.com/assets/7019720/9733839/c456ebce-562e-11e5-9dc9-2dc988f2a8fd.png)
![captura de pantalla 2015-09-08 a las 13 37 04](https://cloud.githubusercontent.com/assets/7019720/9733840/c477eebe-562e-11e5-808d-d66b5cac03cf.png)

**Comment by jjeising** [comment]
> If the image is not loaded, the container will not be shown, otherwise the image will be correctly placed.

I don't think there is any code that depends on the load event of `#imgOverlay`.

> In fact, if you try to adjust the image position inside the load event handler, this event will never be triggered if the image is already loaded.

That's right. Therefor I suggested adding this code (additionally), in case the image isn't loaded at this point. I've debugged the Code, the problem is that in this moment `p = $("#divTimeImageOverlay").height();` returns a height without the image in Safari. This quickfix als works here.

There are multiple better solutions available:

* Change the anchoring of the overlay, anchor it from the bottom instead of from the top. In this way it should also be possible to drop all Javascript code for positioning and just use CSS.
* Add a load handler earlier and do the positioning just there. This may introduce problems, as the load event may not fire for all browsers, e.g. when images are already cached.

But I currently have other issues to solve first and therefor haven't started to explore the first option (and the required cleanup and refactoring for this).

> It should be something related with the adblock.

I can reproduce this here (no issue with Adblock disabled) and it probably has: These extensions use [a special event](https://developer.apple.com/library/safari/documentation/Tools/Conceptual/SafariExtensionGuide/MessagesandProxies/MessagesandProxies.html#//apple_ref/doc/uid/TP40009977-CH14-SW9) to check if loaded content is an ad. This introduces a small delay, enough that the image isn't yet available in the DOM for the height measurement.
